### PR TITLE
[#5846] build(dev): Hive image supports for JDBC SQL standard authorization

### PR DESCRIPTION
**Pull request by roryqi**
### What changes were proposed in this pull request?

Add support for JDBC SQL standard authorization

### Why are the changes needed?

Fix: #5846

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
I tested locally.

### CODE REVIEW

File: docs/docker-image-details.md [hunk 1]
```diff
@@ -168,6 +168,10 @@ Changelog
 You can use this kind of image to test the catalog of Apache Hive.
 
 Changelog
+
+- apache/gravitino-ci:hive-0.1.17
+  - Add support for JDBC SQL standard authorization
+-
```
Comment (xunliu) [inline docs/docker-image-details.md:174]: Please add description of the `hive-site.xml` and `hiveserver2-site.xml` in the here.

Comment (jerqi) [reply docs/docker-image-details.md:175]: Ok, I added refer to other image version records.

File: dev/docker/hive/start.sh [hunk 2]
```diff
@@ -39,6 +39,12 @@ cp -f ${HADOOP_TMP_CONF_DIR}/* ${HADOOP_CONF_DIR}
 cp -f ${HIVE_TMP_CONF_DIR}/* ${HIVE_CONF_DIR}
 sed -i "s/__REPLACE__HOST_NAME/$(hostname)/g" ${HADOOP_CONF_DIR}/core-site.xml
 sed -i "s/__REPLACE__HOST_NAME/$(hostname)/g" ${HADOOP_CONF_DIR}/hdfs-site.xml
+
+if [[ -n "${ENABLE_JDBC_AUTHORIZATION}" ]]; then
```
Comment (xunliu) [inline dev/docker/hive/start.sh:43]: Change `ENABLE_JDBC_AUTHORIZATION` to `ENABLE_SQLBASE_AUTHORIZATION `?

anothe questiong, Will `ENABLE_JDBC_AUTHORIZATION` and `RANGER_HIVE_REPOSITORY_NAME, xxx` conflict??

Comment (jerqi) [reply dev/docker/hive/start.sh:43]: Yes, we they should choose one mode of them.

File: dev/docker/hive/hiveserver2-site-for-jdbc.xml
[the remark this answers is not in the document]
Comment (jerqi) [reply dev/docker/hive/hiveserver2-site-for-jdbc.xml:1]: Renamed. I use `for-sql-base-auth`.

File: dev/docker/hive/hive-site-for-jdbc.xml
[the remark this answers is not in the document]
Comment (jerqi) [reply dev/docker/hive/hive-site-for-jdbc.xml:1]: Renamed. I use `for-sql-base-auth`.

File: dev/docker/hive/start.sh (hunk 2 above)
Comment (jerqi) [reply dev/docker/hive/start.sh:43]: I added a check for it. You can see.

**Pull request merged**